analyzers/src/SonarAnalyzer.Common/Helpers/SonarAnalysisContext.cs

```diff
@@ -92,6 +92,25 @@ internal SonarAnalysisContext(AnalysisContext context, IEnumerable<DiagnosticDes
         public bool ShouldAnalyzeGenerated(Compilation c, AnalyzerOptions options) =>
             ShouldAnalyzeGenerated(context, c, options);
 
+        public bool IsTestProject(Compilation c, AnalyzerOptions options) =>
+            IsTestProject(context.TryGetValue, c, options);
+
+        public static bool IsTestProjectNotCached(Compilation c, AnalyzerOptions options)
```

Comment:
`IsTestProjectNotCached` confuses me. It's like `not TestProject cached`.
```suggestion
        public static bool IsTestProjectNoCache(Compilation c, AnalyzerOptions options)
```